test/CMakeLists.txt

```diff
@@ -135,41 +155,43 @@ if(APPLE)
         set_property(SOURCE "testnativecocoa.m" APPEND_STRING PROPERTY COMPILE_FLAGS " -Wno-error=deprecated-declarations")
     endif()
 elseif(WINDOWS)
-    add_executable(testnative testnative.c testnativew32.c testutils.c)
+    add_sdl_test_executable(testnative NEEDS_RESOURCES testnative.c testnativew32.c testutils.c)
 elseif(HAVE_X11)
-    add_executable(testnative testnative.c testnativex11.c testutils.c)
+    add_sdl_test_executable(testnative NEEDS_RESOURCES testnative.c testnativex11.c testutils.c)
     target_link_libraries(testnative X11)
 endif()
 
-add_executable(testoverlay2 testoverlay2.c testyuv_cvt.c testutils.c)
-add_executable(testplatform testplatform.c)
-add_executable(testpower testpower.c)
-add_executable(testfilesystem testfilesystem.c)
-add_executable(testrendertarget testrendertarget.c testutils.c)
-add_executable(testscale testscale.c testutils.c)
-add_executable(testsem testsem.c)
-add_executable(testsensor testsensor.c)
-add_executable(testshader testshader.c)
-add_executable(testshape testshape.c)
-add_executable(testsprite2 testsprite2.c testutils.c)
-add_executable(testspriteminimal testspriteminimal.c testutils.c)
-add_executable(teststreaming teststreaming.c testutils.c)
-add_executable(testtimer testtimer.c)
-add_executable(testurl testurl.c)
-add_executable(testver testver.c)
-add_executable(testviewport testviewport.c testutils.c)
-add_executable(testwm2 testwm2.c)
-add_executable(testyuv testyuv.c testyuv_cvt.c)
-add_executable(torturethread torturethread.c)
-add_executable(testrendercopyex testrendercopyex.c testutils.c)
-add_executable(testmessage testmessage.c)
-add_executable(testdisplayinfo testdisplayinfo.c)
-add_executable(testqsort testqsort.c)
-add_executable(testbounds testbounds.c)
-add_executable(testcustomcursor testcustomcursor.c)
-add_executable(controllermap controllermap.c testutils.c)
-add_executable(testvulkan testvulkan.c)
-add_executable(testoffscreen testoffscreen.c)
+add_sdl_test_executable(testoverlay2 NEEDS_RESOURCES testoverlay2.c testyuv_cvt.c testutils.c)
+add_sdl_test_executable(testplatform NONINTERACTIVE testplatform.c)
+add_sdl_test_executable(testpower NONINTERACTIVE testpower.c)
+add_sdl_test_executable(testfilesystem NONINTERACTIVE testfilesystem.c)
+add_sdl_test_executable(testrendertarget NEEDS_RESOURCES testrendertarget.c testutils.c)
+add_sdl_test_executable(testscale NEEDS_RESOURCES testscale.c testutils.c)
+add_sdl_test_executable(testsem testsem.c)
+add_sdl_test_executable(testsensor testsensor.c)
+if(NOT EMSCRIPTEN)
+    add_sdl_test_executable(testshader NEEDS_RESOURCES testshader.c)
+endif()
```

Comment:
I disabled building testshader for emscripten because of missing symbols:
```_glColor3f
_glEnd
_glOrtho
_glShadeModel
_glTexCoord2f
_glTexEnvf
_glTranslatef
_glVertex3f
```

Reply:
Hey, not sure if it's the same test, but this is a shader test that is run in Emscripten codebase: https://github.com/emscripten-core/emscripten/blob/89223552124161b1104c698ad45ef7885912db51/test/test_browser.py#L3084

But, it's ran in a browser (instead of nodejs), so may not be trivial to run - it's probably best to keep disabled as you did.

Reply:
This pr is all about getting the tests built.
Current CMake only runs the non-interactive tests on CI, which testshader is not.

I've added `-sLEGACY_GL_EMULATION` to the link arguments for this executable, which got it linking locally.
Thanks for the help!